simphony/io/tests/test_h5_lattice.py

```diff
@@ -73,9 +108,10 @@ def cleanup(self):
         shutil.rmtree(self.temp_dir)
 
     def container_factory(self, name, type_, base_vect, size, origin):
-        return H5Lattice.create_new(
-            self.handle.root, 'foo', type_, base_vect,
-            size, origin, record=CustomRecord)
+        self.group = self.handle.create_group(
```

Comment:
Actually, this is a relevant change. The in-file lattice 'name' parameter is now saved as the name of the subgroup where the lattice is stored. This is analogous how it's done with in-file particles. If the lattice is stored in root, as it was before in the tests, I don't know if the name can be set properly in that case. Also, 'container_factory' does not have a group parameter in it which would point to a coorrect subgroup. Therefore, subgroup has to be created in container_factory, before calling H5Lattice constructor.